source/_integrations/zwave_js.markdown

```diff
@@ -20,56 +20,81 @@ ha_codeowners:
 
-The secure network key is set in the settings for zwave-js-server and
-not in the integration configuration.
+#### 2. The Z-Wave JS to MQTT add-on installed available from the community add-on store
 
-## Current Limitations
+This is the recommended approach if you're running the default version of Home Assistant, including the Supervisor and you like to have maximum control over your Z-Wave network and its nodes.
 
-As this integration is still in the early stages there are some important limitations to be aware of.
+Despite what the name suggests, you can actually run this addon without MQTT enabled. In that case, it will provide you with a full-fledged, attractive and feature-complete UI to manage your Z-Wave nodes and settings.
 
-- Garage door controllers (Barrier Operator CC) are not currently supported.
-- Configuration of Z-Wave nodes and/or configuration with the Home Assistant UI is currently not yet implemented. You will need to use another tool, such as [zwavejs2mqtt](https://github.com/zwave-js/zwavejs2mqtt), to manage device configuration.
-- Polling is currently not supported but may be added in a later release.
-- Support for setting configuration parameters through service calls is currently not supported but may be added in a later release.
-- There currently is no migration path from any of the other Z-Wave implementations in Home Assistant. Your Z-Wave network is however stored on your stick so migrating will only require you to redo your device and entity naming.
+The Z-Wave JS server is **included** in Zwave JS to MQTT, you enable it on the Z-Wave tab within the UI. If you install the add-on from the community repo, this will be done for you automatically.
+
+#### 3. The Z-Wave JS to MQTT docker container
+
+This is the recommended approach if you're running the Home Assistant Core yourself, without the supervisor and it's convenient add-ons. See the [zwavejs2m1tt documentation](https://zwave-js.github.io/zwavejs2mqtt/#/getting-started/quick-start) for instructions. Important note: make sure you run a recent version of the docker image, as that contains the required WS server option.
+
+#### 4. Run the Z-Wave server yourself
+
+This is considered a very advanced scenario where you run the (NodeJS) Z-Wave JS Server (or zwavejs2mqtt) yourself. Installation and maintaining this is out of scope for this document. See the [Z-Wave JS server](https://github.com/zwave-js/zwave-js-server) or [zwavejs2mqtt](https://github.com/zwave-js/zwavejs2mqtt) Github repo's for information.
+
+#### Important note
+
+When you start the Z-Wave JS server, it will begin interviewing your entire Z-Wave network, depending on the size of your network (how many nodes) this can take a while, especially when started for the very first time. Information about your devices is stored in cache files by Z-Wave JS. Be aware that (re)starting the Z-Wave server will cause your network to be (partially) unresponsive until the interview process is done.
```

Comment:
```suggestion
When you start the Z-Wave JS server, it will begin interviewing your entire Z-Wave network. Depending on the size of your network (how many nodes) this can take a while, especially when started for the very first time. Information about your devices is stored in cache files by Z-Wave JS. Be aware that (re)starting the Z-Wave server will cause your network to be (partially) unresponsive until the interview process is done.
```